jobs/link_sender.rb

```diff
@@ -3,57 +3,61 @@
 require_relative '../models/user'
 require_relative '../models/category'
 require_relative '../models/link'
-require 'rufus-scheduler'
 
-def send_links
-  users = User.all
-  users.each do |user|
-    categories = user.categories
-    categories.each do |category|
-      puts category.category_name
-      if category.active
-        bing_web_search = BingService::BingWebSearchApi.new
-        parsed_json = bing_web_search.links(
-          category
-        )
-        result_set = parsed_json['webPages']['value']
-        canonical_url = result_set[0]['displayUrl'].strip
-        unless result_set[0]['displayUrl'].start_with?('http', 'https')
-          canonical_url = 'https://' + result_set[0]['displayUrl'].strip
-        end
-        link_name = category.links.find_by(
-          link: canonical_url)
-        until link_name.nil?
-          result_set.shuffle!
+module LinkSender
+  def send_links
```

Comment:
IMO, this method is too long. Which means that it's doing too much work. Remember **Single Responsibility Principle**...
Several things are happening here:
- Configuring Pony
- Finding a link that hasn't been sent yet
- Saving and sending the link.
etc.